### DISCUSSION TITLE
Correct approach for migrating .server files to vite

### DISCUSSION THREAD

**Discussion by chrizy** [remix-run/remix#9246] [Q&A]
I have a remix cloudflare 2.8.1 project which includes about 40+ .server.ts files that I'm trying to migrate to vite, those files are only called server side from my loaders/actions,  but can't get it to compile  "Server-only module referenced by client".  its seems I need to remove the .server from the filename or use serverOnly$.  that doesn't feel right . am I missing something?   thanks.

**Comment by meza** [comment]
I'm struggling with the same thing.
I've got a few loaders/actions that are server only so they live in .server files now but remix vite:build is complaining that they apparently include themselves? Very strange...

**Comment by meza** [comment]
I don't know if this helps but it helped me: https://github.com/remix-run/remix/discussions/9323#discussioncomment-9240924

**Comment by chrizy** [comment]
Thanks, that might work, but that doesn't quite seem right.  if you look the the remix teams trellix vite demo project, the loaders/actions are in the same file, so thinking there is some other issue at play here.
https://github.com/remix-run/example-trellix/blob/main/app/routes/login/route.tsx

maybe because that demo is not using a server folder or .server files

**Reply by kiliman** [reply]
Imports from _.server_ files should be fine as long as they're used _inside_ a loader or action.

Can you show me what you're doing that's triggering the error?
